### ISSUE TITLE
Lemma 4.3 is missing some condition

### ISSUE THREAD

**Issue by DungMinhDao** [epfml/OptML_course#10]
In Lemma 4.3, the easy direction (Exercise 29) can easily be proved. However, the other direction needs additional conditions, otherwise, it's not correct. For example, consider f(x) = -√x for x ≥ 0. The function and the domain are convex, but there is no subgradient at 0

**Comment by adityavarre** [comment]
True, thanks for the nice catch.  We should change the lemma conditions such that the sub-gradient is non-empty in the interior points of the domain as done in (Theorem 3.1.13 in Nesterov, 2004).  We will update this.
